hogvm/typescript/src/execute.ts

```diff
@@ -2,10 +2,12 @@ import RE2 from 're2'
 
 import { Operation } from './operation'
 import { ASYNC_STL, STL } from './stl/stl'
-import { convertHogToJS, convertJSToHog, getNestedValue, like, setNestedValue } from './utils'
+import { calculateCost, convertHogToJS, convertJSToHog, getNestedValue, like, setNestedValue } from './utils'
 
 const DEFAULT_MAX_ASYNC_STEPS = 100
 const DEFAULT_TIMEOUT_MS = 5000 // ms
+const MAX_MEMORY = 128 * 1024 * 1024 // 128 MB
```

Comment:
This feels like something that should be configurable when executing. 128mb for example is potentially wayyyyyy too high for a cdp function, especially given the state will get serialised to kafka (meaning 500 in a batch * 128mb = 64Gb memory)

Reply:
Yup, but it all runs sequentially, so a batch will never take more than 128MB.

Reply:
I don't want to put this too low because of [this](https://github.com/PostHog/posthog/pull/23564#discussion_r1670623880) inaccuracy in calculation (we might throw too early), however this will be the amount that goes on Kafka in case we pause right now. 🤔

I lowered it to 64MB. We can decrease it even more, make it configurable or introduce a "fat lane". All good for me, but I'd propose to just try as is.

Reply:
Yeah the point is more that this would already explode if we had a function that could generate 64mb of stack memory 🤔 

I guess we can merge, add some way of measuring, create some nasty functions and decide on a much much lower value when we know what a typical function looks like.

Reply:
Before this PR we were running blind, now we get `maxMemSize` in every response `VMState`. Let's track some real world values and see what we need to change. This will already protect us against runaway infinite loops.